### PR TITLE
Use standard temporary directory instead of custom/cache

### PR DESCRIPTION
**Pull request by Lgt2x**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

Temp dir will be `/tmp/Descent3/cache` on Unix. Leave `-tempdir` option anyway, in case we want 2 clients to be run on the same host, so they don't try to lock the same file

### CODE REVIEW

File: Descent3/init.cpp [hunk 1]
```diff
@@ -1988,8 +1988,14 @@ void SetupTempDirectory(void) {
   if (t_arg) {
     strcpy(Descent3_temp_directory, GameArgs[t_arg + 1]);
   } else {
-    // initialize it to custom/cache
-    ddio_MakePath(Descent3_temp_directory, Base_directory, "custom", "cache", NULL);
+    ddio_MakePath(Descent3_temp_directory, std::filesystem::temp_directory_path().u8string().c_str(), "Descent3",
```
Comment (winterheart) [inline Descent3/init.cpp:1991]: Need to use non-throwable version of `temp_directory_path()` and check ec error on path construction.

Comment (Lgt2x) [reply Descent3/init.cpp]: fixed

File: Descent3/init.cpp [hunk 2]
```diff
@@ -1988,8 +1986,19 @@ void SetupTempDirectory(void) {
...
+      Error("Could not find temporary directory: \"%s\"", ec.message().c_str() );
+      exit(1);
+    }
+    ddio_MakePath(Descent3_temp_directory, tempPath.u8string().c_str(), "Descent3",
+                  "cache", NULL);
+    std::filesystem::create_directories(Descent3_temp_directory, ec);
+    if (ec) {
+      Error("Could not create temporary directory: \"%s\"", Descent3_temp_directory);
+      exit(1);
+    }
```
Comment (winterheart) [inline Descent3/init.cpp:2001]: Sorry, another request. This should be outside of `else` block, so we can try to create non-existent directory defined via `t_arg`.

Comment (Lgt2x) [reply Descent3/init.cpp]: done too

**Comment by winterheart** [comment]
Tested on Windows, works fine.

**Pull request merged**